### PR TITLE
Add support for Kubernetes version 1.26

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
This PR does the following:
1. Add support for Kubernetes v1.26
2. Remove support for Kubernetes v1.23
3. Update to the latest minor for supported Kubernetes versions.
4. Remove support for configuring `ConfigSource` in machine spec.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature
/kind cleanup

**Special notes for your reviewer**:
Support for docker will be removed in a separate PR.

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* Add support for Kubernetes v1.26
* Remove support for Kubernetes v1.23
* ConfigSource in Machine specs has been deprecated and is a no-op. Since the dynamic kubelet configuration feature has been removed starting from Kubernetes v1.24.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/apis/cluster/v1alpha1/machine_types.go [hunk 1]
```diff
@@ -89,6 +89,7 @@ type MachineSpec struct {
 	// spec will be automatically copied to the linked NodeRef from the
 	// status. The rest of dynamic kubelet config support should then work
 	// as-is.
+	// Deprecated: This feature has been removed with k8s v1.24.
```
Comment (embik) [inline pkg/apis/cluster/v1alpha1/machine_types.go:92]: Can we add a note that setting this does nothing?

**Comment by ahmedwaleedmalik** [comment]
- Amazon Linux is failing for AWS
- Flatcar is failing on Azure

**Comment by ahmedwaleedmalik** [comment]
/test pull-machine-controller-e2e-azure

Flake

File: pkg/userdata/rockylinux/testdata/kubelet-v1.24-aws.yaml [hunk 2]
```diff
@@ -241,6 +241,8 @@ write_files:
       --lock-file=/tmp/kubelet.lock \
       --container-runtime=remote \
       --container-runtime-endpoint=unix:///run/containerd/containerd.sock \
+      --dynamic-config-dir=/etc/kubernetes/dynamic-config-dir \
+      --feature-gates=DynamicKubeletConfig=true \
```
Comment (kron4eg) [inline pkg/userdata/rockylinux/testdata/kubelet-v1.24-aws.yaml:245]: This shouldn't happen, DynamicKubeletConfig been deleted in 1.24

Comment (ahmedwaleedmalik) [reply pkg/userdata/rockylinux/testdata/kubelet-v1.24-aws.yaml]: I somehow messed up the constraint 🤦 Updated the PR.

**Comment by ahmedwaleedmalik** [comment]
I've requested an override for `pull-machine-controller-e2e-aws-arm` so that we can get this PR merged. It's failing due to capacity issues:

> error InsufficientInstanceCapacity: We currently do not have sufficient a1.medium capacity in the Availability Zone you requested (eu-central-1a)

I'm going to test `pull-machine-controller-e2e-aws-arm` with my upcoming PRs.

**Comment by xrstf** [comment]
/override pull-machine-controller-e2e-aws-arm

AWS AZ is out of capacity.

**Comment by mfranczy** [comment]
Also flatcar is failing on KubeVirt. I see that from the log that KubeVirt tests that time were disabled. I will take a look to see what's wrong with.

**Comment by mfranczy** [comment]
Looks like it's connected to the thing that on our flatcar image we have containerd 1.5.4, probably it should be 1.6, meaning we have to update the flatcar image.

**Pull request merged**